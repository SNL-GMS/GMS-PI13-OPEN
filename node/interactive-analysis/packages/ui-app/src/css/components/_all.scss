@import 'about-menu';
@import 'body';
@import 'list';
@import 'login';
@import 'quadra-grid';
@import 'workspace';
