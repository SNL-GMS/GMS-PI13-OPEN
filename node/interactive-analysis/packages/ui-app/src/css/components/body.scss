body.bp3-dark {
  background-color: $gms-background;
}

.bp3-navbar {
  padding: 0;
}

.bp3-control input:focus ~ .bp3-control-indicator {
  outline: none;
  outline-offset: 0px;
}
.bp3-control {
  margin-bottom: 0px;
}
.bp3-control:not(.checkbox-horizontal) + .bp3-control:not(.checkbox-horizontal) {
  margin-top: 10px;
}
.bp3-control.checkbox-horizontal + .bp3-control.checkbox-horizontal {
  margin-left: 8px;
}
.bp3-dark .bp3-control.bp3-switch .bp3-control-indicator {
  background: $gms-popover-background;
  margin-left: 4px;
}
.bp3-dark .bp3-control.bp3-switch .bp3-control-indicator::before {
  background: $gms-main;
}
.bp3-dark .bp3-control.bp3-switch:hover .bp3-control-indicator {
  background: $gms-soft;
}

.bp3-dark .bp3-button:not([class*='bp3-intent-']) {
  background: $gms-button;
}
.bp3-dark .bp3-button:not([class*='bp3-intent-']):hover {
  background: $gms-button-hover;
}
.bp3-dark .bp3-button.bp3-minimal {
  background: none;
  background-color: none;
}

.bp3-dark .bp3-popover .bp3-popover-content,
.bp3-popover.bp3-dark .bp3-popover-content {
  background-color: $gms-popover-background;
  color: $gms-prominent;
}

.bp3-dialog {
  padding-bottom: 0.5rem;
}
.bp3-dark .bp3-dialog {
  background-color: $gms-dialog-background;
}

.bp3-dialog-header {
  box-shadow: none;
}

.bp3-dark .bp3-dialog-header {
  box-shadow: none;
}
*::-webkit-scrollbar {
  width: 10px;
  height: 10px;
}

*::-webkit-scrollbar-track {
  background-color: $gms-scrollbar-background;
  width: 10px;
}

*::-webkit-scrollbar-thumb {
  background: $gms-scrollbar;
  border-radius: 2px;
}

:focus {
  outline: none;
  outline-offset: 0px;
}

// override some focus styles from blueprint that dont look good.
.bp3-control input:focus ~ .bp3-control-indicator {
  outline: none;
  outline-offset: 0px;
}

.max {
  position: absolute;
  top: 0px;
  right: 0px;
  bottom: 0px;
  left: 0px;
}

.duo-grid-left {
  position: absolute;
  top: 22%;
  right: 0%;
  left: 1.5%;
  width: 48%;
  height: 18%;
}

.duo-grid-right {
  position: absolute;
  top: 22%;
  right: 0%;
  left: 52%;
  width: 48%;
  height: 18%;
}

.bp3-menu {
  min-width: auto !important;
}

.gms-disconnected {
  position: absolute;
  top: 0px;
  right: 0px;
  bottom: 0px;
  left: 0px;
  z-index: 100;
  background-color: rgba($gms-background, 0.85);
  display: flex;
  justify-content: center;
  align-items: center;
}
