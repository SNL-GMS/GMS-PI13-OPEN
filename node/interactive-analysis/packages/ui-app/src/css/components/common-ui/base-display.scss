.base-display {
  display: flex;
  position: relative;
  flex-direction: column;
  &.full-width-height {
    height: 100%;
    width: 100%;
    flex-grow: 1;
  }
}
