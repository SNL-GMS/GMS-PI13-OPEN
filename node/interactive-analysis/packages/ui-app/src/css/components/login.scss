.login-screen-body {
  background-color: $gms-prominent-background;
  height: 100%;
}

.login-container {
  font-size: 16px;
  width: 100%;
  line-height: 32px;
  display: flex;
  align-items: center;
  height: 100%;
  flex-direction: column;
  justify-content: center;
}

.login-row {
  display: flex;
  width: 380px;
  align-items: flex-start;
  margin-top: 12px;
}

.login-row.login-button {
  justify-content: flex-end;
}

.login-label {
  margin-right: 16px;
  width: 84px;
}

.login-input {
  border-radius: 0;
  color: $analyst-requires-review;
  background-color: $gms-prominent-background;
  font-size: 16px;
  width: 280px;
  line-height: 16px;
  margin-top: 3px;
  border: none;
  border-bottom: 1px solid $gms-soft;
}

.login-legal {
  max-width: 660px;
  line-height: 12px;
  font-size: 12px;
  text-align: justify;
}

.app-menu-button {
  padding-right: 1.25em;
  padding-left: 1em;
  border-left: 1px solid $gms-prominent;
  border-radius: 0px;
}
