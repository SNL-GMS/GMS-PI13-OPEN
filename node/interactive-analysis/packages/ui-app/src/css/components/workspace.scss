.workspace-container {
  display: flex;
  height: 100%;
  width: 100%;
  --webkit-user-select: none;
  flex-direction: column;
}

.bp3-navbar.workspace-navbar {
  background-color: $gms-popover-background;
  height: 35px;
  display: flex;
  justify-content: space-between;
}

.workspace-logo {
  filter: invert(100%);
  margin-left: 9px;
  padding: 6px 3px;
}

.workspace-logo__label {
  margin-left: 0.25rem;
}
.keypress-signifier {
  animation-name: flash-active;
  animation-duration: 0.5s;
  animation-timing-function: ease-out;
}
@keyframes flash-active {
  from {
    color: $gms-input-highlight;
  }
  to {
    color: $gms-main;
  }
}

.workspace-invalid-state {
  display: flex;
  flex: 1 1 auto;
  height: 100%;
  width: 100%;
  max-width: 100%;
  position: relative;
}

.workspace {
  width: calc(100%);
  max-width: 100%;
  height: calc(100% - 36px);
  flex: 1 1 auto;
}

.workspace-navbar__image-label {
  display: flex;
  align-items: center;
}

.unopened-layout {
  margin-left: 24px;
}

// Override the default menu header. To be safe, we add our own class, too.
.bp3-dark .menu-title > h6,
.bp3-dark .bp3-menu-header > h6 {
  color: darken($gms-prominent, 5%);
  letter-spacing: 1px;
}
