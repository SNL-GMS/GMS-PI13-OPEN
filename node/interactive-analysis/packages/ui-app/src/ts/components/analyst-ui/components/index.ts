export { ReduxApolloWaveformDisplay as WaveformDisplay } from './waveform-display';
export { Events } from './events';
export { History } from './history';
export { SignalDetections } from './signal-detections';
export { Workflow } from './workflow';
export { ReduxApolloMap as Map } from './map';
export { Location } from './location';
export { Magnitude } from './magnitude';

export { ReduxApolloAzimuthSlowness as AzimuthSlowness } from './azimuth-slowness/azimuth-slowness-container';
