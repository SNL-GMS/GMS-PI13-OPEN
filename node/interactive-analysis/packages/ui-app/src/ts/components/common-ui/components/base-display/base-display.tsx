import { getDataAttributesFromProps, useElementSize, useForceUpdate } from '@gms/ui-util';
import defer from 'lodash/defer';
import * as React from 'react';
import { BaseDisplayContext } from './base-display-context';
import { BaseDisplayProps } from './types';

const useForceUpdateOnFirstRender = () => {
  const forceUpdate = useForceUpdate();
  React.useEffect(() => {
    defer(forceUpdate);
  }, []);
};

/**
 * A base display that should be at the base of all display components.
 * Adds consistent padding to each display, and exposes the width and height
 * of the display in the BaseDisplayContext.
 * @param props requires a reference to the glContainer.
 * Also accepts data attributes in the form 'data-cy': 'example-component'
 */
export const BaseDisplay: React.FunctionComponent<React.PropsWithChildren<
  BaseDisplayProps
>> = props => {
  /**
   * Base display size behavior
   */
  const [displayRef, heightPx, widthPx] = useElementSize();

  /**
   * On the very first mount, call forceUpdate so that height and width will
   * propagate to the consumers of the context.
   */
  useForceUpdateOnFirstRender();

  /**
   * the context menu handler, if provided
   */
  const { onContextMenu } = props;

  /**
   * Get any data attributes provided to this display (like data-cy attributes)
   */
  const dataAttributes = getDataAttributesFromProps(props);

  return (
    <div
      className={`base-display ${props.className ?? ''}`}
      ref={ref => (displayRef.current = ref)}
      onContextMenu={onContextMenu}
      {...dataAttributes}
    >
      <BaseDisplayContext.Provider
        value={{
          glContainer: props.glContainer,
          widthPx: widthPx || (props.glContainer?.width ?? 0),
          heightPx: heightPx || (props.glContainer?.height ?? 0)
        }}
      >
        {props.children}
      </BaseDisplayContext.Provider>
    </div>
  );
};
