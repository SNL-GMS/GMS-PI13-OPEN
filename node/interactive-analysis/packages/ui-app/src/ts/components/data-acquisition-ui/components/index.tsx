export { StatusConfiguration } from './status-configuration';
export { StationConfiguration } from './station-configuration';
export { StationInformation } from './station-information';
export { TransferGaps } from './transfer-gaps';
export { ConfigureStationGroups } from './configure-station-groups';
export { SohOverview } from './soh-overview';
export { StationStatistics } from './station-statistics';
export { SohEnvironment } from './soh-environment';
export { SohEnvironmentHistory } from './environment-history';
export { SohLagHistory } from './soh-lag-history';
export { SohMissingHistory } from './soh-missing-history';
export { SohLag, SohMissing, SohTimeliness } from '~data-acquisition-ui/components/soh-bar-chart';
