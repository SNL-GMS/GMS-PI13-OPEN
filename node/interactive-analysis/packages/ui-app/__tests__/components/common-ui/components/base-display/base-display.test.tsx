import * as React from 'react';
import { BaseDisplay } from '../../../../../src/ts/components/common-ui/components/base-display';

// tslint:disable-next-line:no-implicit-dependencies no-var-requires no-require-imports
const Enzyme = require('enzyme');

// set up window alert and open so we don't see errors
(window as any).alert = jest.fn();
(window as any).open = jest.fn();

// tslint:disable-next-line: deprecation
const lodash = require.requireActual('lodash');
lodash.uniqueId = () => '1';

describe('System Messages Display', () => {
  it('should be defined', () => {
    expect(BaseDisplay).toBeDefined();
  });
  const base: any = Enzyme.mount(
    <BaseDisplay
      glContainer={
        {
          widthPx: 150,
          heightPx: 150,
          on: jest.fn()
        } as any
      }
      className="mock-display"
    />
  );

  it('matches snapshot', () => {
    expect(base).toMatchSnapshot();
  });
});
