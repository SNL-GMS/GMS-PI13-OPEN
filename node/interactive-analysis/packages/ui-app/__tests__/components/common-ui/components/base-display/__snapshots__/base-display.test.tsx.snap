// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`System Messages Display matches snapshot 1`] = `
<Component
  className="mock-display"
  glContainer={
    Object {
      "heightPx": 150,
      "on": [MockFunction],
      "widthPx": 150,
    }
  }
>
  <div
    className="base-display mock-display"
  />
</Component>
`;
